internal/id/id.go

```diff
@@ -1,38 +1,59 @@
 package id
 
 import (
-	"sync"
+	"sync/atomic"
 )
 
-var mu sync.Mutex
-var turn, cycle, counter uint32 = 0, 0, 0
-
-var totalBits uint32 = 32
-var turnBits uint32 = 4
-var counterBits = totalBits - turnBits
+const (
+	totalBits   uint32 = 32                                // Total bits for the ID
+	turnBits    uint32 = 4                                 // Bits for turn (allows up to 16 turns before wrap-around
+	workerBits  uint32 = 4                                 // Bits allocated for worker ID (supports up to 16 workers, 2^4)
```

Comment:
Isn't this way too small? This would mean that the system can only support up to 16 threads and won't guarantee unique IDs beyond that limit.

This is a serious limitation, we should at least support up to 128 or 256 cores, only then can we truly scale the performance with the hardware.